### ISSUE TITLE
checksec script bug found (fix provided) [NameError: global name 'dt_rpath' is not defined]

### ISSUE THREAD

**Issue by matrix1001** [Gallopsled/pwntools#1090]
# checksec script bug
bug script
`pwnlib/elf/elf.py`
bug function
```python
    @property
    def runpath(self):
        """:class:`bool`: Whether the current binary has a ``RUNPATH``."""
        dt_runpath = self.dynamic_by_tag('DT_RUNPATH')

        if not dt_runpath:
            return None

    return self.dynamic_string(dt_rpath.entry.d_ptr)
```
problem: on some occasion, cause this problem. and checksec failed
```
NameError: global name 'dt_rpath' is not defined
```
influence: all modules that use checksec
# fix advice
```python
    @property
    def runpath(self):
        """:class:`bool`: Whether the current binary has a ``RUNPATH``."""
        dt_runpath = self.dynamic_by_tag('DT_RUNPATH')

        if not dt_runpath:
            return None

    return self.dynamic_string(dt_runpath.entry.d_ptr)
```
apparently it is due to some carelessness.

**Comment by zachriggle** [comment]
This is a duplicate of #1008 which has, admittedly, been sitting stagnant for too long.  

It looks like it wasn't merged yet because of CI checks required to merge to `stable`.

**Comment by Arusekk** [comment]
Closed in #1008
